### ISSUE TITLE
Open web address not working

### ISSUE THREAD

**Issue by amenolo** [Ylianst/MeshCentral#1146]
Lately I noticed that in some of my clients the `open web address` and `open chat` (not surprisingly since in fact as we all know retrieves a specific web address) commands fail to run:

![web](https://user-images.githubusercontent.com/29974944/79075185-834c4e00-7cf1-11ea-964d-6f4cd9064edc.gif)

At the moment it happens in two PMs with Windows 10 updated to last patches along with different user configurations/programs, and seems to be browser-independent.

Server is running MC v0.5.1-o, reinstalling agents didn't change a thing.

Do you think could be worth a look?
In case, which logs may I provide in order to help?

Thanks

**Comment by krayon007** [comment]
Are these machines joined to a domain? I know the mechanism i use to spawn a process into the user session, doesn't seem to work with domain joined users. I'm still trying to find a way to get that working.

**Comment by amenolo** [comment]
Actually are not, they're single user machines in  different domestic LAN.

**Comment by krayon007** [comment]
If you go to the terminal tab, and right click on connect, and connect user shell, does it work? (This uses the same dispatch mechanism as the chat thing.)

**Comment by amenolo** [comment]
You're right, not working as well:

![terminal](https://user-images.githubusercontent.com/29974944/79076201-957dba80-7cf8-11ea-85c8-b9aaca13fddb.gif)

**Comment by krayon007** [comment]
The root terminal works though, correct? if so I'll get back to you with some commands I want you to try in the root terminal to see if we can get this working.

**Comment by amenolo** [comment]
> The root terminal works though, correct?

Affirmative.

> I'll get back to you with some commands I want you to try

You have my attention!

**Comment by krayon007** [comment]
ok, from the root terminal, type in the following commands:
SCHTASKS /CREATE /F /TN TestTask /SC ONCE /ST 00:00 /RU xxx /TR "C:\Windows\System32\cmd.exe /C START http://www.google.com"

(That's all on one line... And replace xxx with the username)

Then do:
SCHTASKS /RUN /TN TestTask

That should popup the default browser to www.google.com.
I'm guessing that won't work, because that's basically what the dispatcher does... I don't specify the domain when I specify the username, so I'm thinking on your system, you may have to. If so, try entering in the username, as  domain\username.

You can delete the task with:
SCHTASKS /DELETE /F TN TestTask

**Comment by amenolo** [comment]
Tests done, maybe something interesting found: in one of the reported client (the other one's off at the moment) I have to declare username parameter along with its own role otherwise I get an error:

```
C:\Program Files\Mesh Agent>SCHTASKS /CREATE /F /TN TestTask /SC ONCE /ST 03:00 /RU Plex /TR "C:\Windows\System32\cmd.exe /C START http://www.google.com"
RROR: The parameter is incorrect.
(40,4):UserId:
```

This way I'm able to create the task...

```
C:\Program Files\Mesh Agent>SCHTASKS /CREATE /F /TN TestTask /SC ONCE /ST 03:00 /RU Plex\Administrator /TR "C:\Windows\System32\cmd.exe /C START http://www.google.com"
SUCCESS: The scheduled task "TestTask" has successfully been created.
```
...that once executed leads to the well expected outcome: no browser popup.

On the other hand in a client where the dispatcher works fine, to avoid sintax error I have to use the only username as parameter of /RU option:

```
C:\Program Files\Mesh Agent>SCHTASKS /CREATE /F /TN TestTask /SC ONCE /ST 03:00 /RU Test\Administrator /TR "C:\Windows\System32\cmd.exe /C START http://www.google.com"
ERROR: No mapping between account names and security IDs was done.
```
```
C:\Program Files\Mesh Agent>SCHTASKS /CREATE /F /TN TestTask /SC ONCE /ST 03:00 /RU Test /TR "C:\Windows\System32\cmd.exe /C START http://www.google.com"
SUCCESS: The scheduled task "TestTask" has successfully been created.
```
Please note in both cases the users are native local administrator and I can't see any rough differences in their configurations (which I managed back then).

At last out of curiosity I tried with a brand new account (at first as normal user then upgraded to local administrator) added to the client with the issue and...all is working fine (and the /RU sintax to use in manual test requires the username only)! :astonished:

Maybe after all is not about a MC bug...I'm lost here :confused:

**Comment by flyRV** [comment]
I appear to be experiencing the same issue as I found here during my research but perhaps have some additional light to shed on the issue.

In my testing, I've been able to replicate this issue using a domain user _administrator@domain.com,_ and I was able to resolve this problem by renaming the the local/built-in _administrator_ user. Once I renamed this local administrator user, the chat browser session launched successfully.

As already discussed, the issue presents itself differently depending on the logged in user at the remote endpoint, whereby it works with some domain users that are logged but not for others. For those that do not work, I ran through the scheduled tasks test provided and found that I had to specify the user format as **domain\user**, at which point running the task via terminal would properly launch a web browser. I believe this is exactly what is reported here, but just being verbose to ensure nothing is lost in translation. 

    SCHTASKS /CREATE /F /TN TestTask /SC ONCE /ST 00:00 /RU domain\user /TR "C:\Windows\System32\cmd.exe /C START http://www.google.com"

At this point, I believe the root of the issue is both in how Mesh possibly spawns this process but also in how Windows deals with this action when a local user exists whose name matches that of a domain user. In my experience, Windows will default to using a local user when the FQDN or _domain\user_ format is not explicitly used.

As one additional testing measure, I set the local/built-in _administrator_ user password to be the same as the domain _administrator@domain.com_ but the process would still not spawn/no chat window launched until I renamed this local/built-in user.
